staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go

```diff
@@ -45,6 +48,8 @@ type Config struct {
 
 	Codec  runtime.Codec
 	Copier runtime.ObjectCopier
+	// Transformer allows the value to be transformed prior to persisting into etcd.
+	Transformer value.Transformer
```

Comment:
Yes, working with others to do that (proposal is still being discussed for that part).  The transformer interface should be opaque to config and vice versa.